src/traffic_layout/engine.cc

```diff
@@ -565,83 +543,99 @@ from_group(std::string const &user, gid_t group_id, gid_t path_gid)
       return true;
     }
   }
-#endif
   return false;
 }
 
 // set permission to the map in verify runroot
 static void
-set_permission(std::vector<std::string> const &dir_vector, RunrootMapType &path_map, RunrootMapType &permission_map,
-               RunrootMapType &usergroup_map, std::string &user)
+set_permission(PermissionMapType &permission_map, std::string const &user)
 {
-  // active group and user of the path
-  std::pair<gid_t, uid_t> giduid = get_giduid(user);
+  bool new_line = false;
 
-  gid_t ts_gid = giduid.first;
-  uid_t ts_uid = giduid.second;
+  passwd *pwd = getpwnam(user.c_str());
+  if (!pwd) {
+    ink_fatal("missing password database entry for '%s'", user.c_str());
```

Comment:
Are we sure this is the correct message?

Reply:
It means we can't find that username. I think this message is not bad. Any other suggestions?